src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights.AspNetCore/TargetingHttpContextMiddleware.cs

```diff
@@ -50,7 +47,12 @@ public async Task InvokeAsync(HttpContext context, ITargetingContextAccessor tar
 
             if (targetingContext != null)
             {
-                context.Items[TargetingIdKey] = targetingContext.UserId;
+                var requestTelemetry = context.Features.Get<RequestTelemetry>();
+
+                if (requestTelemetry != null)
+                {
+                    requestTelemetry.Properties.Add(Constants.TargetingIdKey, targetingContext.UserId);
```

Comment:
Before calling `Add` method, we need to do `ContainsKey` check.

Reply:
I suppose the question to me is- if the targeting id already exists, what should we do? I think there's 3 options:
1. Throw
2. Do nothing
3. Override

Depending on what we want- I think the right functions are:
1. `.Add`
2. `.TryAdd` or `ContainsKey`
3. `[...] = ...`

I could see an argument for all of them. Although I think Override might actually make the most sense to ensure our TargetingId tagging is reliable and if people add the middleware twice it's not breaking.

Reply:
I don't think the middleware should do anything if the TargetingId is already there. We shouldn't just blindly override it. With that, I wouldn't even call `targetingContextAccessor.GetContextAsync()` if the TargetingId already exists.

Reply:
Went ahead and adjusted this to `[...]=...` for now.